corehq/apps/geospatial/views.py

```diff
@@ -216,9 +228,47 @@ def page_url(self):
 
     @property
     def page_context(self):
-        return {
+        page_context = {
             'mapbox_access_token': settings.MAPBOX_ACCESS_TOKEN,
         }
+        page_context.update(self._case_filters_context())
+        return page_context
+
+    def _case_filters_context(self):
+        # set up context for report filters template to be used for case filtering
+        return {
+            'report': {
+                'title': self.page_name,
+                'section_name': self.section_name,
+                'show_filters': True,
+            },
+            'report_filters': [
+                dict(field=f.render(), slug=f.slug) for f in self.filter_classes
+            ],
+            'report_filter_form_action_css_class': CSS_ACTION_CLASS,
+        }
+
+    @property
+    @memoized
+    def filter_classes(self):
+        # copied from corehq.apps.reports.generic.GenericReportView.filter_classes
```

Comment:
this duplicate is avoided in a later commit by extracting the implementation in a function